cmd/server.go

```diff
@@ -86,7 +88,7 @@ func generateOpenApiSpec() []byte {
 	var securitySchemas map[string]interface{}
 	err := json.Unmarshal(securitySchemasData, &securitySchemas)
 	if err != nil {
-		log.Panicf("Logging err: %s\n", err.Error()) // panic if there is an error
+		log.Error("Logging err", slog.Any("error", err.Error())) // panic if there is an error
```

Comment:
Becuase previously we used `Panicf`, do we want to [change it](https://github.com/tink3rlabs/todo-service/pull/29#discussion_r1697896973)?
Also, the inline comment does not match.
@deanefrati

Reply:
Slog  doesn't have `fatal or panic`

wherever we have `log.Panic` or `log.Fatal`, I have used

```
slog.Error
os.Exit(1)
```

Reply:
Since we're creating our own logger package, wouldn't it be easier to add a function to it that does this instead of having to use os.Exit every time we want this functionality?  We could create a Fatal function in the logger package that would log the error and exit.

Generally speaking, i've tried to use fatal/panic when we're trying to instantiate something and that fails since that's really unrecoverable, if we keep running and just log the error there would be some functionality that we tried to use that does't work.

Reply:
I did think about creating a function, we have our own logger package but I am using `slog.SetDefault` to centrally manage logging configurations without the need to redefine or instantiate logger objects across your codebase since slog already does it for us.

if we create a new logging function then we need to maintain and extend this function whenever slog introduces new features or parameters, such as additional log formatting options, we need to update our Fatal function parameters to incorporate those changes. I feel it's easier to use os.exit()

Reply:
Not sure i understand... we can also chat about it if you have time but what would we need to maintain if we add a function like the following to our logger package:

func Fatal(message string, err error) {
		logger.Error("Logging err", slog.Any("error", err.Error()))
		os.Exit(1)
}

Reply:
created separate function